src/molecular-weight.jl

```diff
@@ -0,0 +1,105 @@
+# DNA Adenine                 dAMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/12599
+# RNA Adenine                 AMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6083
+# DNA Cytosine                dCMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/13945
+# RNA Cytosine                CMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6131  
+# DNA Guanine                 dGMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/1135398597
+# RNA Guanine                 GMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/135398631
+# DNA Thymine                 dTMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/9700
+# Calculations following the following guidelines: https://ymc.eu/files/imported/publications/556/documents/YMC-Expert-Tip---How-to-calculate-the-MW-of-nucleic-acids.pdf
+
+# Creating the arrays DNA_WEIGHTS and RNA_WEIGHTS to list all weights. In ambigous cases, let's list the string "Undefined", so if called, it returns an error
+# The values already account for loss of water = - 18.02
+DNA_WEIGHTS = [0, 313.2, 289.18, "Undefined", 329.2, "Undefined", "Undefined", "Undefined", 304.19, "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined"]
+
+RNA_WEIGHTS = [0, 329.2, 305.18, "Undefined", 345.2, "Undefined", "Undefined", "Undefined", 306.16, "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined"]
+
+# Defining a function that accepts nucleotide sequences (either DNA or RNA) and calculate the molecular weight in g/mol
+# Note the additional keywords alphabet, five_terminal_state and strand_number, which have default cases, if not specified
+# In case of using double strand, the five_terminal_state of the complmentary strand is assumed to be the same as the input strand
+# Function only accepts double as strand_number for DNA sequences
+function molecular_weight(nucseq::NucSeq, alphabet=:DNA, five_terminal_state=:hydroxyl, strand_number=:single)
+    weight = 0
+    if alphabet == :DNA
+      for dnucleotide in nucseq
+        dna_weight = DNA_WEIGHTS[reinterpret(UInt8, dnucleotide) + 1]
+        weight += dna_weight
+        end
+    elseif alphabet == :RNA
+      for nucleotide in nucseq
+        rna_weight = RNA_WEIGHTS[reinterpret(UInt8, nucleotide) + 1]
+        weight += rna_weight
+        end 
+    else 
+        throw(ArgumentError("Unknown Alphabet. Must be :DNA or :RNA"))
+    end
+    if five_terminal_state == :hydroxyl
+        weight =  weight - 62
+    elseif five_terminal_state == :phosphate
+        weight = weight + 79
+    elseif five_terminal_state == :triphosphate 
+        weight = weight + 178
+    else
+        throw(ArgumentError("Unknown five_terminal_state. Must be :hydroxyl, :phosphate or :triphosphate"))  
+    end
+    if strand_number == :single
+        return round(weight; digits = 3)
+    elseif strand_number == :double && alphabet == :DNA
```

Comment:
A better way to do it is just to call `molecular_weight` with the same arguments, except changing the strand to `single`